pyhap/hap_server.py

```diff
@@ -895,93 +37,36 @@ def create_hap_event(cls, bytesdata):
-        logger.debug("Current connections %s", self.connections)
-        sock = self.connections.pop(client_addr, None)
-        if sock is not None:
-            self._close_socket(sock)
-        if not isinstance(exception, socket.timeout) \
-                and exception.errno not in self.TIMEOUT_ERRNO_CODES:
-            raise exception
-
-    def get_request(self):
-        """Calls the super's method, caches the connection and returns."""
-        client_socket, client_addr = super(HAPServer, self).get_request()
-        logger.info("Got connection with %s.", client_addr)
-        self.connections[client_addr] = client_socket
-        return (client_socket, client_addr)
-
-    def finish_request(self, request, client_address):
-        """Handle the client request.
-
-        HAP connections are not closed. Once the client negotiates a session,
-        the connection is kept open for both incoming and outgoing traffic, including
-        for sending events.
-
-        The client can gracefully close the connection, but in other cases it can just
-        leave, which will result in a timeout. In either case, we need to remove the
-        connection from ``self.connections``, because it could also be used for
-        pushing events to the server.
-        """
-        try:
-            self.RequestHandlerClass(request, client_address,
-                                     self, self.accessory_handler)
-        except (OSError, socket.timeout) as e:
-            self._handle_sock_timeout(client_address, e)
-            logger.debug('Connection timeout')
-        except Exception as e:
-            logger.debug('finish_request: %s', e, exc_info=True)
-            raise
-        finally:
-            logger.debug('Cleaning connection to %s', client_address)
-            conn_sock = self.connections.pop(client_address, None)
-            if conn_sock is not None:
-                self._close_socket(conn_sock)
-
-    def server_close(self):
-        """Close all connections."""
-        logger.info('Stopping HAP server')
-
-        # When the AccessoryDriver is shutting down, it will stop advertising the
-        # Accessory on the network before stopping the server. At that point, clients
-        # can see the Accessory disappearing and could close the connection. This can
-        # happen while we deal with all connections here so we will get a "changed while
-        # iterating" exception. To avoid that, make a copy and iterate over it instead.
-        for sock in list(self.connections.values()):
-            self._close_socket(sock)
+        self.server = None
+        self._serve_task = None
+
+    async def async_start(self, loop):
+        """Start the http-hap server."""
+        self.server = await loop.create_server(
+            lambda: HAPServerProtocol(loop, self.connections, self.accessory_handler),
+            self._addr_port[0],
+            self._addr_port[1],
+        )
+
+    def async_stop(self):
```

Comment:
I am a bit confused about the method name and not being async. Can you elaborate? Saw something similar in the accessory driver as well.

Reply:
The method must be called in the event loop and not from the executor as it is not thread safe.

Aka it's only safe to call in async context, but since nothing is being awaited it doesn't need to be a coroutine. 

Usually in home assistant we decorate these with @callback

Reply:
Looks like we have `@callback`.  I'll add that now

Reply:
Ah ok, makes sense now.